sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAdministrationAsyncClient.java

```diff
@@ -473,8 +487,259 @@ private AttestationToken generatePolicySetToken(String policy, AttestationSignin
+                    });
+            });
+    }
+
+    /**
+     * Sets the current policy for an attestation type.
+     *
+     * @param options Options for this API call, encapsulating both the X.509 certificate to add to the set of policy
+     *               signing certificates and the signing key used to sign the request to the service.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response to an attestation policy operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<PolicyCertificatesModificationResult>> addPolicyManagementCertificateWithResponse(PolicyManagementCertificateOptions options) {
+        return withContext(context -> addPolicyManagementCertificateWithResponse(options, context));
+    }
+
+    /**
+     * Adds a new attestation policy certificate to the set of policy management certificates.
+     *
+     * @param options Options for this API call, encapsulating both the X.509 certificate to add to the set of policy
+     *               signing certificates and the signing key used to sign the request to the service.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response to an attestation policy operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PolicyCertificatesModificationResult> addPolicyManagementCertificate(PolicyManagementCertificateOptions options) {
+        return addPolicyManagementCertificateWithResponse(options)
+            .flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Adds a new policy management certificate to the set of policy management certificates.
+     *
+     * @param options Options for this API call, encapsulating both the X.509 certificate to add to the set of policy
+     *               signing certificates and the signing key used to sign the request to the service.
+     * @param context Context for the operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response to an attestation policy operation.
+     */
+    Mono<Response<PolicyCertificatesModificationResult>> addPolicyManagementCertificateWithResponse(PolicyManagementCertificateOptions options, Context context) {
+        Objects.requireNonNull(options.getCertificate());
+        Objects.requireNonNull(options.getAttestationSigner());
```

Comment:
Since this is a requirement for both add and remove calls should be disallow creating options type with these values as null?

Reply:
This was fixed in my code which added certificate and signer as ctor arguments.